### PR TITLE
Create testing script documentation

### LINKED ISSUE #113: Document and publish the scripts, approach, and scope
Referring to the automated testing scripts in https://github.com/Quansight-Labs/jupyter-a11y-mgmt/issues/95

TODO:
- Open a PR that will go into jupyter/accessibility 
- For the `scope` we should align with the discussions we mentioned in #97 of what we test vs what not (i.e. the content generated by the author themselves)

### PR DESCRIPTION
**Pull request by isabela-pf**
Fixes [quansight-labs/jupyter-a11y-mgmt #113](https://github.com/Quansight-Labs/jupyter-a11y-mgmt/issues/113).

We've been discussing testing scripts for converting manual testing tasks into automated, JupyterLab-specific tests. A part of these discussions have been having a common language for us to convert these tasks along to make them as clear as possible for turning into code. This PR
- Adds documentation of existing JupyterLab testing script proposals
- Adds a template for testing scripts for future contribution

## Questions

- [x] Where should this new information go? It's in the `testing`directory for now, but I'm unclear if that's confusing or not since it is not a test that runs, but rather support for turning into tests that run. Answer: leave it here for now and find the appropriate place in repo reorganization.
- [x] Is now a good time to add the [ACT-rules](https://act-rules.github.io/pages/about) listing, or are we sticking with WCAG for now because it is more familiar? Answer: this can wait until another work cycle.
- [x] What do y'all think is missing? Answer: cross linking between template and in-use work (added).

Thanks in advance for the review!

### CODE REVIEW

File: testing/scripts/jupyterlab-testing-scripts.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# JupyterLab Testing Scripts
+
+This is a series of [testing scripts](). designed to mimic common manual accessibility tests in an automated testing setting for JupyterLab. They currently reference [WCAG 2.1](), but ideally will be updated for future versions of WCAG or to align with [ACT-rules]().
```
Comment (trallard) [inline testing/scripts/jupyterlab-testing-scripts.md]: There are multiple missing links

Comment (trallard) [inline testing/scripts/jupyterlab-testing-scripts.md]:
```suggestion
This is a series of [testing scripts]() designed to mimic common manual accessibility tests in an automated testing setting for JupyterLab. They currently reference [WCAG 2.1](), but ideally will be updated for future versions of WCAG or to align with [ACT-rules]().
```

File: testing/scripts/testing-script-template.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+| 3.  | | 
+| 4.  | |
+| 5.  | |
+| 6.  | |
+| 7.  | |
+| 8.  | |
+| 9.  | | 
+| 10. | | 
+
+Scripts can have as many or as few steps as needed to complete the task. Some steps may not need expected behavior, but try to add expected behavior as consistently as possible.
```
Comment (trallard) [inline testing/scripts/testing-script-template.md]:
```suggestion
Scripts can have as many or as few steps as needed to complete the task. Some steps may not need expected behavior, but try to add expected behavior as consistently as possible.

```

File: testing/scripts/jupyterlab-testing-scripts.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+#### Proposed testing script
+
+| Step | Expected Behavior | 
+|-----|-----|
+| 1. Open default JupyterLab | JupyterLab opens with unmodified workspace. | 
+| 2. Open the file menu with a mouse click | File menu opens and full list of menu items appears. |
+| 3. Navigate to menu item New Launcher with arrow keys | Focus moves through the File menu list items until it reaches the New Launcher item. | 
+| 4. Use touch screen input to create new Notebook from Launcher | The New Notebook from Launcher is selected and the command is initiated. |
+| 5. Success if new notebook opens |  |
```
Comment (trallard) [inline testing/scripts/jupyterlab-testing-scripts.md]:
```suggestion
| 5. Success if new notebook opens |  |
```

**Review by trallard — CHANGES_REQUESTED** [review]
Some minor comments here and there - but I really like the flow and conciseness of the scripts + the template.

File: testing/scripts/testing-script-template.md (hunk 2 above)
Comment (isabela-pf) [reply testing/scripts/testing-script-template.md]: Is this missing a comment, or am I missing the suggestion? I don't see what I'm supposed to fix.

File: testing/scripts/jupyterlab-testing-scripts.md (hunk 1 above)
Comment (isabela-pf) [reply testing/scripts/jupyterlab-testing-scripts.md]: Thank you for catching that! I'm really embarrassed by this one. 🙃

File: testing/scripts/jupyterlab-testing-scripts.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+#### Proposed testing script
+
+| Step | Expected Behavior | 
+|-----|-----|
+| 1. Open default JupyterLab | JupyterLab opens with unmodified workspace. | 
+| 2. Open the file menu with a mouse click. | File menu opens and full list of menu items appears. |
+| 3. Navigate to menu item New Launcher with arrow keys. | Focus moves through the File menu list items until it reaches the New Launcher item. | 
+| 4. Use touch screen input to create new Notebook from Launcher. | The New Notebook from Launcher is selected and the command is initiated. |
+| 5. Success if new notebook opens. |  |
```
Comment (trallard) [inline testing/scripts/jupyterlab-testing-scripts.md:100]:
```suggestion
| 5. Success if new notebook opens. |  |

```

end of file line

File: testing/scripts/jupyterlab-testing-scripts.md (hunk 1 above)
Comment (trallard) [reply testing/scripts/jupyterlab-testing-scripts.md]: it happens 🤗

File: testing/scripts/testing-script-template.md (hunk 2 above)
Comment (trallard) [reply testing/scripts/testing-script-template.md]: it is marked as outdated so I assume the suggestion is not applicable all good now

**Comment by trallard** [comment]
> Is now a good time to add the [ACT-rules](https://act-rules.github.io/pages/about) listing, or are we sticking with WCAG for now because it is more familiar?

I think the wording you have is enough for now - we might want to add this ACT discussion as part of our upcoming planning - note we also have to plan for alignment with WCAG 2.2.

> What do y'all think is missing
Maybe a note at the end of the doc to mention we have a template and x-link to it


For now we can leave this in the `testing` directory as I plan to continue with the reorganisation of the repo in the upcoming sprint

**Review by trallard — APPROVED** [review]
Minor comments - nothing absolutely necessary I think

**Comment by trallard** [comment]
Merged @isabela-pf  - can you open a PR from this repo master to jupyter/a11y:master - can leave open for a few days and merge since it has been reviewed here 🎉

**Comment by isabela-pf** [comment]
This actually brings up a question I was planning to ask in our meeting today! 

Since many things have been merged into `quansight-labs/accessibility:master`, am I supposed to open basically one big PR with this _and_ the workshop links _and_ multiple PRs worth of testing updates? Or is there a super git skill that I don't know to break up the work in one branch? 

Or do I just do multiple PRs from the fork's specific branches `quansight-labs/accessibility:testingscripts`?

I feel like the big PR would be hard to review on `jupyter/accessibility`.

**Pull request merged**